shell/platform/darwin/ios/framework/Source/FlutterDartProjectTest.mm

```diff
@@ -102,14 +102,30 @@ - (void)testFLTAssetsURLFromBundle {
     id mockBundle = OCMClassMock([NSBundle class]);
     id mockMainBundle = OCMPartialMock([NSBundle mainBundle]);
     NSString* resultAssetsPath = @"path/to/foo/assets";
-    OCMStub([mockBundle pathForResource:@"flutter_assets" ofType:@""]).andReturn(nil);
-    OCMStub([mockMainBundle pathForResource:@"flutter_assets" ofType:@""])
+    OCMStub([mockBundle pathForResource:@"Frameworks/App.framework/flutter_assets" ofType:@""])
+        .andReturn(nil);
+    OCMStub([mockMainBundle pathForResource:@"Frameworks/App.framework/flutter_assets" ofType:@""])
         .andReturn(resultAssetsPath);
     NSString* path = FLTAssetsPathFromBundle(mockBundle);
     XCTAssertEqualObjects(path, @"path/to/foo/assets");
   }
 }
 
+- (void)testFLTAssetPathReturnsTheCorrectValue {
+  {
+    // Found asset path in info.plist
+    id mockBundle = OCMClassMock([NSBundle class]);
+    OCMStub([mockBundle objectForInfoDictionaryKey:@"FLTAssetsPath"]).andReturn(@"foo/assets");
+    XCTAssertEqualObjects(FLTAssetPath(mockBundle), @"foo/assets");
+  }
+  {
+    // No asset path in info.plist, use default value
+    id mockBundle = OCMClassMock([NSBundle class]);
+    OCMStub([mockBundle objectForInfoDictionaryKey:@"FLTAssetsPath"]).andReturn(nil);
+    XCTAssertEqualObjects(FLTAssetPath(mockBundle), kDefaultAssetPath);
+  }
+}
+
```

Comment:
These tests are both of a private helper method; there's nothing about the structure here that would indicate that someone making a change like yours couldn't just change these tests if they broke.

The deeper problem is that we appear to have no tests of the public `lookupKeyForAsset:*` methods in this file. Could you add tests for each variant of that, with comments on the expectations saying that they are testing public API behaviors and anything that breaks the expectations is likely to break plugins?